rust/template/ddlog.h

```diff
@@ -954,6 +954,45 @@ extern size_t ddlog_get_strlen(const ddlog_record *rec);
  */
 extern const char * ddlog_get_str_with_length(const ddlog_record *rec, size_t *len);
 
+/*
+ * Create a serialized record from a NULL-terminated string `t` containing
+ * the serialization scheme and a NULL-terminated string `s` containing
+ * the serialized data.
+ * This function copies `s` to an internal buffer, so the caller is
+ * responsible for deallocating `s` if it was dynamically allocated.
+ *
+ * Returns `NULL` if `t` or `s` are not a valid null-terminated UTF8 strings.
+ */
+extern ddlog_record* ddlog_serialized(const char *t, const char *s);
+
+/*
+ * Create a serialized record.
+ *
+ * `t` - points to the start of a UTF8 string containing the
+ *       serialization scheme. The string does not have to be
+ *       NULL-terminated. The pointer must not be `NULL`
+ * `t_len` - length of string in bytes.
+ * `s` - points to the start of a UTF8 string containing the
+ *       serialization scheme.  The string does not have to be
+ *       NULL-terminated. The pointer must not be `NULL`, unless
+ *       `s_len==0`, in which case the function ignores the value of the
+ *       pointer and returns a record containing an empty string.
+ * `s_len` - length of string in bytes.
+ *
+ * This function copies `s` and `t` to internal buffers, so the caller is
```

Comment:
Ahh, whoops, I was copying it originally, then decided I would probably always know what kind of serialisation scheme a dataset was using.

That's it corrected now.

Reply:
As to the use-case, I'm toying around with a service that takes a differential-ddlog program as a payload, and wires up it's input and output tables to some external streams using some additional metadata.

Deserializing in datalog would result in a lot of unnecessary memory overhead since most of the payload data would never get used.

Reply:
DDlog-as-a-service would be neat.  If you feel that any of the stuff you build can be reused, a PR would be most welcome :)